libs/partners/upstage/langchain_upstage/chat_models.py

```diff
@@ -0,0 +1,101 @@
+import os
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+)
+
+import openai
+from langchain_core.pydantic_v1 import Field, SecretStr, root_validator
+from langchain_core.utils import (
+    convert_to_secret_str,
+    get_from_dict_or_env,
+)
+from langchain_openai import ChatOpenAI
+
+
+class ChatUpstage(ChatOpenAI):
+    """ChatUpstage chat model.
+
+    To use, you should have the environment variable `UPSTAGE_API_KEY`
+    set with your API key or pass it as a named parameter to the constructor.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_upstage import ChatUpstage
+
+
+            model = ChatUpstage()
+    """
+
+    @property
+    def lc_secrets(self) -> Dict[str, str]:
+        return {"upstage_api_key": "UPSTAGE_API_KEY"}
+
+    @classmethod
+    def get_lc_namespace(cls) -> List[str]:
+        return ["langchain", "chat_models", "upstage"]
+
+    @property
+    def lc_attributes(self) -> Dict[str, Any]:
+        attributes: Dict[str, Any] = {}
+
+        if self.upstage_api_base:
+            attributes["upstage_api_base"] = self.upstage_api_base
+
+        return attributes
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "upstage-chat"
+
+    model_name: str = Field(default="solar-1-mini-chat", alias="model")
+    """Model name to use."""
+    upstage_api_key: Optional[SecretStr] = Field(default=None, alias="api_key")
+    """Automatically inferred from env are `UPSTAGE_API_KEY` if not provided."""
+    upstage_api_base: Optional[str] = Field(
+        default="https://api.upstage.ai/v1/solar", alias="base_url"
```

Comment:
if this is encoded in the SDK as well, I would prefer to make this default to `None` (and let the sdk set the default)

Reply:
Good point! We use the OpenAI SDK, so its default value is OpenAI's API endpoint. That is why we set the default value here.